pkg/download/addons/with_pool.go

```diff
@@ -64,13 +65,13 @@ func (p *withpool) List(ctx context.Context, mod string) ([]string, error) {
 	return vers, nil
 }
 
-func (p *withpool) Info(ctx context.Context, mod, ver string) ([]byte, error) {
+func (p *withpool) Info(ctx context.Context, mod, ver string, lggr log.Entry) ([]byte, error) {
```

Comment:
We don't need to update the function signatures all over, because a `log.Entry` is already existent in the Context map. All you have to do is call `log.EntryFromContext(ctx)` and you will get the Entry. If the Entry is not there, you'll get a NoOp logger so that we never panic.